packages/firestore/src/util/obj_map.ts

```diff
@@ -107,4 +107,64 @@ export class ObjectMap<KeyType extends Equatable<KeyType>, ValueType> {
   isEmpty(): boolean {
     return objUtil.isEmpty(this.inner);
   }
+
+  [Symbol.iterator](): Iterator<{ key: KeyType; value: ValueType }> {
+    // We don't care about the keys, all of the actual keys are in the
+    // arrays that are the values of the inner object.
+    const it = objUtil.values(this.inner)[Symbol.iterator]();
+    return new ObjectMapIterator<KeyType, ValueType>(it);
+  }
+}
+
+/**
+ * Implements an Iterator over ObjectMap.
+ */
+export type IteratorEntry<KeyType, ValueType> = {
```

Comment:
FWIW, This is the Entry type from SortedMap.

Reply:
Yup, but I don't think a dependency between them is appropriate.